### ISSUE TITLE
[java] UnnecessaryFullyQualifiedName fails to recognize illegal self reference in enums

### ISSUE THREAD

**Issue by Adam-** [pmd/pmd#4631]
**Affects PMD Version:** 6.44.0

**Rule:** [UnnecessaryFullyQualifiedName](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname)

**Description:**

UnnecessaryFullyQualifiedName fails to recognize illegal self reference with enum constructors

**Code Sample demonstrating the issue:**

https://github.com/runelite/runelite/blob/36979d67f7a0ba420ec3ed3189ac27ff3c1a1abb/runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/Tree.java#L53

```java
import api.ObjectID;
import static api.ObjectID.*;

public enum {
  TEAK_TREE(Duration.of(15, GAME_TICKS), ObjectID.TEAK_TREE, TEAK_TREE_36686, TEAK_TREE_40758),
```

This errors because there is a wildcard import above `import static net.runelite.api.ObjectID.*;`, however removing `ObjectID.` causes an illegal self-reference to the enum.

**Expected outcome:**

PMD doesn't report this as an error.

**Running PMD through:** maven

**Comment by jsotuyod** [comment]
I think this is a duplicate of #1790 which has already been fixed in PMD 7. This should be rechecked with PMD 7.0.0-rc3 to validate.
